tutorials/3d/spring_arm.rst

```diff
@@ -0,0 +1,117 @@
+
+.. image:: img/spring_arm_children.svg
+
+Spring arm with a camera
+------------------------
+
+When a camera is placed as a child of a spring arm, a pyramid representing the camera will be used as the shape.
+
+This pyramid represents the **near plane** of the camera:
+
+.. image::: img/spring_arm_camera_shape.svg
+
+.. note:: If the spring arm is given a specific shape, then that shape will **always** be used.
+
+    The camera's shape is only used if the camera is a **direct child** of the spring arm.
+
+    If no shape is provided and the camera is not a direct child, the spring arm will fall back to using a ray cast which is inaccurate for camera collisions and not recommended.
+
+Every physics process frame, the spring arm will perform a motion cast to check if anything is collided with:
+
+.. image:: img/spring_arm_camera_motion_cast.svg
+
+When the shape hits something, the camera will be placed at the collision point:
+
+.. image:: img/spring_arm_camera_collision.svg
+
+Setting up the spring arm and camera
+------------------------------------
+
+Let's add a spring arm camera setup to the platformer demo.
+
+.. note:: You can download the Platformer3D demo on `GitHub <https://github.com/godotengine/godot-demo-projects/tree/master/3d/platformer>`_ or using the `Asset Library <https://godotengine.org/asset-library/asset/2748>`_.
+
+In general, for a third-person camera setup, you will have three nodes as children of the node that you're following:
+
+- `Node3D` (the "pivot point" for the camera)
+
+    - `SpringArm3D`
+
+        - `Camera3D`
+
+Set these up as children of our player and give them unique names so we can find them in our script:
+
+.. image:: img/spring_arm_editor_setup.png
+
+Let's move the pivot point up by `1.5` on the Y-axis so that it's not on the ground:
+
+.. image:: img/spring_arm_pivot_setup.png
+
+
+Give the spring arm a length of `3` so that it is placed behind the character:
+
+.. image:: img/spring_arm_length_setup.png
+
+.. note:: Leave the "Shape" of the spring arm as `<empty>`. This way, it will use the camera's pyramid shape.
+
+    If you want, you can also try other shapes - a sphere is a common choice since it slides smoothly along edges.
+
+Update the top of `player.gd` to grab the camera and the pivot points by their unique names:
+
+.. code-block:: gdscript
+
+    @onready var _camera := %Camera3D as Camera3D
+    @onready var _camera_pivot := %CameraPivot as Node3D
+
+Add an `_unhandled_input` function to check for camera movement and then rotate the pivot point accordingly:
+
+.. code-block:: gdscript
+
+    func _unhandled_input(event: InputEvent) -> void:
+        if event is InputEventMouseMotion:
+            _camera_pivot.rotation.x += event.relative.y * 0.01
+            _camera_pivot.rotation.y += -event.relative.x * 0.01
```

Comment:
Similarly, this code didn't work until I deleted the existing camera node in the player scene. I think it was blocking the unhandled input somehow?

You may also want to add a one-liner for vertical clamping code. The result of small tutorials like this doesn't need to be production-ready, but it should avoid obvious errors like allowing 360 degree unclamped motion in a character controller:

https://github.com/user-attachments/assets/8f04d7d0-6c3c-4fc2-885f-2e658fb0800c

Reply:
Added a note about removing the existing camera and made the screenshot a bit larger so you can see all the nodes in the scene.

Also, good call on the clamping logic 👍